src/filehash.ts

```diff
@@ -1,23 +1,37 @@
 import * as fs from 'fs';
-import * as child_process from 'child_process';
+import * as glob from 'glob';
+import {createHash} from 'crypto';
 
 import normalizePath from './normalizePath';
 
+const sha256sum = (filename: string) =>
+  createHash('sha256').update(fs.readFileSync(filename)).digest('hex');
+
+// const sha256sumWithPath = (filename: string) =>
+//   `${sha256sum(filename)} ?${filename}`
+// // ^ NOTE sha256sumWithPath is used simply to keep backwards compat. with older versions that used find+xargs+shasum
+
 /** Calculate a sha256 hash of a file or directory. */
-export default (path0: string, format: 'hex'|'base64' = 'hex') => {
+export default (path0: string, format: 'hex' | 'base64' = 'hex') => {
   const path = normalizePath(path0);
   // this assumes local files/dirs TODO validate that
   if (!fs.existsSync(path)) {
     throw new Error(`Invalid path ${path} for filehash`);
   }
   const isDir = fs.lstatSync(path).isDirectory();
-  const shasumCommand = 'shasum -p -a 256';
-  const hashCommand = isDir
-    ? `find ${path} -type f -print0 | xargs -0 ${shasumCommand} | ${shasumCommand}`
-    : `${shasumCommand} ${path}`;
 
-  const result = child_process.spawnSync(hashCommand, [], {shell: true});
-  const hexHash = result.stdout.toString().trim().split(' ')[0];
+  let hexHash: string;
+  if (isDir) {
+    // Ray, do you prefer this:
+    // hexHash = createHash('sha256').update(
+    //   glob.sync(`${path}/**`, {nodir: true}).map(sha256sumWithPath).join("\n"),
+    // ).digest('hex');
+    // or this:
+    hexHash = createHash('sha256').update(glob.sync(`${path}/**`, {nodir: true}).map(sha256sum).join(",")).digest('hex');
```

Comment:
Wow this is hard to read 😑 

The difference from the above commented out lines just the join character?

Is it possible to be backwards compatible with the old hash mechanism (I assume that is what `sha256sumWithPath` is)? (My main concern with backwards compatibility is with template approval, but it turns out it uses its own [hash function](https://github.com/unbounce/iidy/blob/master/src/cfn/approvedTemplateVersionLocation.ts#L17).)